python/pyspark/serializers.py

```diff
@@ -213,7 +213,15 @@ def __repr__(self):
         return "ArrowSerializer"
 
 
-def _create_batch(series):
+def _create_batch(series, copy=False):
```

Comment:
Do we need `copy` here?
I might miss something but looks like all occurrence of `copy=copy` in this method are always copied by `s.fillna(0)` in advance so we don't need to use `copy=True`.

Reply:
Right, I forgot that `fillna` returns a copy.  Do you think it would be worth it to first check for any nulls and only `fillna` if needed?  The mask of nulls is already created so just need to add a function like this in `_create_batch`:

```
def fill_series_nulls(s, mask):
    return s.fillna(0) if mask.any() else s
```
What do you think @ueshin ?

Reply:
Hmm, I guess it depends.
With the method, it can reduce the number of copy if `s` doesn't include null values, but also it might increase the number if `s` includes null values and `copy=True`.

Reply:
Yeah, we don't want to end up double copying if `copy=True`.  Let me try something and if it ends up making things too complicated then we can remove the copy flag altogether and just rely on `fillna(0)` to always make a copy - not ideal but will be more simple

Reply:
@ueshin this ended up having no effect, so I took it out.  For the case of Timestamps, the timezone conversions will make a copy regardless.  For the case of ints being promoted to floats then that means they will have null values and need to call `fillna(0)` which makes a copy anyway.  So it seems this only makes copies when necessary.